internal/abci/abci.go

```diff
@@ -685,14 +681,71 @@ func (a *AbciApp) prepareMempoolTxns(txs [][]byte, maxBytes int, log *log.Logger
 
 		if bytes.Equal(tx.Sender, proposerAddr) {
 			proposerNonce = tx.Body.Nonce + 1
-		}
+			propTxs = append(propTxs, tx)
+		} else {
+			// Append the transaction to the final list.
+			if _, ok := otherTxns[string(tx.Sender)]; !ok {
+				otherTxns[string(tx.Sender)] = make([]*indexedTxn, 0)
+				senders = append(senders, string(tx.Sender))
 
-		finalTxns = append(finalTxns, txs[tx.is])
+			}
+			otherTxns[string(tx.Sender)] = append(otherTxns[string(tx.Sender)], tx)
+		}
 		nonces = append(nonces, tx.Body.Nonce)
 		i++
 	}
 
-	return finalTxns, proposerNonce
+	// truncate based on our max block size since we'll have to set
+	// ConsensusParams.Block.MaxBytes to -1 so that we get ALL transactions even
+	// if it goes beyond max_tx_bytes.  See:
```

Comment:
We haven't yet set it to -1 AFAIK,

Reply:
Yeah, it was in TODO, I removed it by mistake.